### PR TITLE
Add support for Apollo Guidance Computer extension

### PR DESCRIPTION
**Pull request by chrislgarry**

### CODE REVIEW

**Comment by larsbrinkhoff** [comment]
Hello @chrislgarry,

Please read the [contribution guidelines](https://github.com/github/linguist/blob/master/CONTRIBUTING.md) for how to add support for a new programming language.

Thank you!

**Comment by Alhadis** [comment]
Haha! I was thinking about adding this earlier today. :D

Ordinarily, AGC wouldn't be widespread enough to qualify for addition to the site's recognised languages... but c'mon, this is a piece of human history, and a cornerstone of mankind's technological achievements. You think we could make an exception, @arfon, @pchaigno? Pleeeeease?

DON'T MAKE ME BUST OUT THE PUPPY DOG EYES.

**Comment by larsbrinkhoff** [comment]
Hey @Alhadis, no one said it doesn't qualify.  You just need to provide an estimate of how many repositories (or files) there are.

I'd say there may be about 25000 Apollo Guidance Computer files.

**Comment by Alhadis** [comment]
Well, this is a historical repository, so it wouldn't surprise me if it were a one-off thing... there might be other NASA repos with similar legacy code perhaps, I haven't checked. But since BCPL has seen broader use, I would've assumed AGC wouldn't qualify.

@chrislgarry There are some preliminary tasks that need to be performed on a Linguist branch before a language can be properly added. They're documented [very clearly here](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language)... although since there isn't a dedicated grammar for syntax highlighting, I might have to write one. =D

**Comment by larsbrinkhoff** [comment]
There's not just one repository.  There seem to be more than 70 copies of [virtualagc](https://github.com/search?q=virtualagc&ref=searchresults&type=Repositories) alone.

**Comment by Alhadis** [comment]
That'll teach me for not bothering to look. 👍 I regret nothing.

**Comment by Alhadis** [comment]
@arfon If this gets merged, I promise to write another hand-spun language grammar just to get AGC's code highlighted properly on GitHub. It'll be a tremendous honour for a space-loving nerd like myself.

And by "space-loving", I mean the region that encapsulates our planetary horizons. Because tabs > spaces.

**Comment by arfon** [comment]
> I promise to write another hand-spun language grammar just to get AGC's code highlighted properly on GitHub.

Alright @Alhadis - how quickly do you think you could do this? It would be awesome to add support for AGC detection and highlighting simultaneously...

**Comment by Alhadis** [comment]
Well if I rushed, I could probably knock it out in a couple days, but I enjoy taking the time to research and write solid, tested grammars thoroughly. Although this is assembly (and I can't exactly see it having the complexity of, say, Haskell), it'd still involve some measure of double-checking and stuff, which isn't something I'd prefer to rush.

Basically, I was planning on submitting a PR after it got merged. =)

**Comment by Alhadis** [comment]
Although... wait, when were you thinking of cutting the next release of Linguist? Sometime pretty soon?

**Comment by Alhadis** [comment]
Never mind, who am I kidding? I could get it done in less than a day, just not to the level of detail I'm used to (and prefer doing).

I'll start tomorrow.

**Comment by arfon** [comment]
> Although... wait, when were you thinking of cutting the next release of Linguist? Sometime pretty soon?

For this, I will cut a new Gem... 💎 💎 💎

File: samples/Assembly/BURN_BABY_BURN--MASTER_IGNITION_ROUTINE.agc [hunk 1]
```diff
@@ -0,0 +1,1059 @@
+# Copyright:	Public domain.
```
Comment (pchaigno) [inline samples/Assembly/BURN_BABY_BURN--MASTER_IGNITION_ROUTINE.agc]: @chrislgarry This sample needs to be in a `samples/AGC/` directory ;)

**Comment by pchaigno** [comment]
Should the name be `Apollo Guidance Computer` or `AGC`? I guess it needs to be `Apollo Guidance Computer` if we want people to be able to search for it using the `apollo` keyword...

**Comment by arfon** [comment]
Let's go for the latter (so people can search)

**Comment by Alhadis** [comment]
Should it be labelled `Apollo-11 Guidance Computer` or just `Apollo Guidance Computer`...?

**Comment by chrislgarry** [comment]
John, it should be Apollo Guidance Computer. Ill move the sample to its own
directory. Thanks. Sorry Im at work now so slow to respond.

**Comment by Alhadis** [comment]
Ace, thanks. I'll update the grammar's name accordingly.

(Yes, it's 3:38 am and I'm starting a new grammar... I did say I'd start **tomorrow**)

**Comment by chrislgarry** [comment]
Thanks a lot everyone.

**Comment by Alhadis** [comment]
@arfon, I forgot how syntactically simplistic assembly is. I'm damned if this will take more than a couple hours.

Sorry, I guess I lost my sense of scale since I was in the middle of [tying up this grammar](https://github.com/Alhadis/language-roff/commit/cfb7d72335f8264cd9f5dcef8e571b03afe6c44a) when I answered, heh.

**Comment by chrislgarry** [comment]
Guys, I messed up the PR by trying to use the web interface because I don't have command line access right now. I made a new one: https://github.com/github/linguist/pull/3112. Is it possible to consolidate to preserve the discussion.

**Pull request closed without merging**